packages/manager-components/src/utils/price.formatter.tsx

```diff
@@ -0,0 +1,182 @@
+import React from 'react';
+import { OsdsText } from '@ovhcloud/ods-components/react';
+import { ODS_TEXT_COLOR_HUE, ODS_TEXT_SIZE } from '@ovhcloud/ods-components';
+import { ODS_THEME_COLOR_INTENT } from '@ovhcloud/ods-common-theming';
+import {
+  OvhSubsidiary,
+  IntervalUnitType,
+  OVH_SUBSIDIARY_CURRENCY,
+} from '../enumTypes';
+import './translations/translations';
+
+export interface PriceOptions {
+  ovhSubsidiary: OvhSubsidiary;
+  locale: string;
+  tFunction: (key: string) => string;
+}
+
+const getPrice = (value: number, tax?: number): number => {
+  const valueWithTax = tax ? value + tax : value;
+  return valueWithTax / 100000000;
+};
+
+const convertIntervalPrice = (
+  price: number,
+  intervalUnit: IntervalUnitType,
+): number => {
+  switch (intervalUnit) {
+    case IntervalUnitType.day:
+      return price / 365;
+    case IntervalUnitType.month:
+      return price / 12;
+    case IntervalUnitType.year:
+      return price;
+    case IntervalUnitType.none:
+      return price;
+    default:
+      return price;
+  }
+};
+
+const getPriceTextFormatted = (
+  ovhSubsidiary: OvhSubsidiary,
+  locale: string,
+  priceValue: number,
+): string => {
+  try {
+    return new Intl.NumberFormat(locale.replace('_', '-'), {
+      style: 'currency',
+      currency: OVH_SUBSIDIARY_CURRENCY[ovhSubsidiary],
+      minimumFractionDigits: 2,
+      maximumFractionDigits: 2,
+    }).format(priceValue);
+  } catch (e) {
+    const fallbackLocale = 'fr_FR';
+    return new Intl.NumberFormat(fallbackLocale.replace('_', '-'), {
+      style: 'currency',
+      currency: OVH_SUBSIDIARY_CURRENCY[ovhSubsidiary],
+      minimumFractionDigits: 2,
+      maximumFractionDigits: 2,
+    }).format(priceValue);
+  }
+};
+
+const TextPriceContent: React.FC<{ children: React.ReactNode }> = ({
+  children,
+}) => (
+  <OsdsText
+    size={ODS_TEXT_SIZE._200}
+    color={ODS_THEME_COLOR_INTENT.default}
+    hue={ODS_TEXT_COLOR_HUE._500}
+    className="ml-1"
+  >
+    {children}
+  </OsdsText>
+);
+
+export const formatPrice = (
```

Comment:
if it returns a ReactNode it should be a component

Reply:
Sure but it returns a ReactNode so create a component instead of a function.

If you want it to be an utils function you have to split the logic that format the price (and returns a string) and the logic that displays the price (and returns a ReactNode).
Right now it does both : format and display.